pkg/controller/controllername/controllername.go

```diff
@@ -0,0 +1,48 @@
+package controllername
+
+import (
+	"strings"
+	"unicode"
+)
+
+const (
+	metricsNameDelimiter = "_"
+	loggerNameDelimiter  = "-"
+)
+
+type ControllerNamer interface {
+	MetricsName() string
+	LoggerName() string
+}
+
+// controllerName ex. {"My","Controller", "Name"} ->  MyControllerName
+type controllerName []string
+
+func NewControllerName(name []string) controllerName {
+	cn := make(controllerName, len(name))
+
+	for i, w := range name {
+		cn[i] = strip(strings.ToLower(w))
+
+	}
+	return cn
+}
+
+// strip removes spaces and non letters
+func strip(s string) string {
```

Comment:
can you add unit tests for this strip function as well?

Reply:
yes